### PR TITLE
Switch to provider engine

### PR DESCRIPTION
**Pull request by ptescher**
Also remove Web3 (dapps should provide their own)

### CODE REVIEW

File: rollup.config.ts [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import commonjs from 'rollup-plugin-commonjs';
+import image from 'rollup-plugin-img';
+import json from 'rollup-plugin-json';
```
Comment (jtescher) [inline rollup.config.ts]: do we still need the rollup plugins?

**Comment by pixelmatrix** [comment]
I'm concerned about doing this without extensive testing. Can we just remove the dependency on web3 for now?

**Comment by pat2man** [comment]
@pixelmatrix Yeah, I'll mark as WIP.

**Comment by ptescher** [comment]
@pixelmatrix @jtescher did more testing, this looks good to go.

**Comment by pixelmatrix** [comment]
How are we looking on test coverage?

File: package.json [hunk 2]
```diff
@@ -2,56 +2,43 @@
   "name": "bitski",
   "description": "Bitski JavaScript SDK",
   "license": "MIT",
-  "main": "dist/bitski.umd.js",
-  "module": "dist/bitski.es5.js",
+  "main": "dist/lib/bitski.js",
   "typings": "dist/types/bitski.d.ts",
   "repository": {
     "type": "git",
-    "url": "https://github.com/BitskiCo/bitski-js"
```
Comment (pixelmatrix) [inline package.json]: this should be /bitski-js

File: docs/README.md [hunk 3]
```diff
@@ -23,7 +23,7 @@ npm install --save bitski
 Alternatively you can add this script tag to your app’s `<head>`:
 
 ```html
-<script src="https://cdn.jsdelivr.net/npm/bitski@0.1.9/lib/bitski.js"></script>
+<script src="https://cdn.jsdelivr.net/npm/bitski@0.2.0/lib/bitski.js"></script>
```
Comment (pixelmatrix) [inline docs/README.md]: Do we need to include more detail here now that web3 is unbundled?

File: package.json [hunk 4]
```diff
@@ -2,56 +2,43 @@
...
+    "url": "https://github.com/BitskiCo/bitski-js-sdk"
   },
-  "version": "0.1.10",
+  "version": "0.2.0",
```
Comment (pixelmatrix) [inline package.json]: do we want to bump the version yet? or should we wait until we do a release branch?

File: package.json [hunk 5]
```diff
@@ -2,56 +2,43 @@
...
   "scripts": {
     "lint": "tslint 'src/**/*.ts'",
-    "build": "tsc && rollup -c rollup.config.ts && webpack --mode production && typedoc",
-    "prepack": "npm run build",
-    "dev": "rollup -c rollup.config.ts -w",
+    "build": "tsc && typedoc",
+    "prepack": "npm install && tsc",
```
Comment (pixelmatrix) [inline package.json]: curious about these build scripts. so we're no longer using webpack or rollup?

Comment (ptescher) [reply package.json]: @pixelmatrix I switched to vanilla tsc for debugging, and found it to work a lot better. Node seems to take care of these things automatically.

File: package.json (hunk 4 above)
Comment (ptescher) [reply package.json]: Good point, I will revert.

File: docs/README.md (hunk 3 above)
Comment (ptescher) [reply docs/README.md]: I don't know if anyone really needs CDN stuff.

**Comment by ptescher** [comment]
@jtescher This should be good enough for a beta release

**Pull request merged**